engine/resource/provider/FilesystemResourceProvider.cpp

```diff
@@ -5,16 +5,74 @@
 #include <utility>
 #include <resource/Resource.h>
 
+#if defined(__APPLE__) && defined(__MACH__)
+    #include "CoreFoundation/CoreFoundation.h"
+#endif
+
 using namespace chira;
 
+CHIRA_CREATE_LOG(FILESYSTEM);
+
 FilesystemResourceProvider::FilesystemResourceProvider(std::string path_, bool isPathAbsolute, const std::string& name_)
     : IResourceProvider(name_)
     , path(std::move(path_))
     , absolute(isPathAbsolute) {
     FilesystemResourceProvider::nixifyPath(this->path);
     this->path = String::stripRight(this->path, '/');
-    if (!this->absolute)
+    if (!this->absolute) {
+    // MacOS is a lot more complex. There's more than 1 place our resources could be at
+    // One requires a bit of work to access though.
+#if defined(__APPLE__) && defined(__MACH__)
+        // Finding the bundle and it's resources directory
+        std::string status = "";
+        CFBundleRef mainbundle =  CFBundleGetMainBundle();
+        if (mainbundle) {
```

Comment:
nit, I find this cleaner since the variable only exists in the `if` but comes down to personal preference
```suggestion
        if (CFBundleRef mainbundle =  CFBundleGetMainBundle()) {
```